src/theory/bv/theory_bv_rewrite_rules_simplification.h

```diff
@@ -1557,6 +1557,40 @@ Node RewriteRule<ShiftZero>::apply(TNode node) {
 
 /* -------------------------------------------------------------------------- */
 
+/**
+ * UgtUrem
+ *
+ * (bvugt (bvurem T x) x)
+ *   ==>  (ite (= x 0_k) (bvugt T x) false)
+ *   ==>  (and (=> (= x 0_k) (bvugt T x)) (=> (not (= x 0_k)) false))
+ *   ==>  (and (=> (= x 0_k) (bvugt T x)) (= x 0_k))
+ *   ==>  (and (bvugt T x) (= x 0_k))
+ *   ==>  (and (bvugt T 0_k) (= x 0_k))
+ */
+
+template <>
+inline bool RewriteRule<UgtUrem>::applies(TNode node)
+{
+  return (node.getKind() == kind::BITVECTOR_UGT
```

Comment:
I believe that this rule is only valid if `bv-div-zero-const` is `false` because otherwise a modulus of zero invalidates the rule:

```
(declare-const x (_ BitVec 4))                                               
(declare-const y (_ BitVec 4))                                               
(assert (not (= (bvugt (bvurem y x) x) (ite (= x #x0) (bvugt y #x0) false))))
(check-sat)                                                                  
```

returns `sat` with `--no-bv-div-zero-const`. The rule should only apply if that option is set to `true`.

Reply:
That's right. I added this check, as well as two regressions.